lib/signup_screen.dart

```diff
@@ -38,86 +91,182 @@ class SignUpPageState extends State<SignUpPage> {
+                // the functions are passed from here as stepNumber variable is defined in this class
+                buttonClickHandler: () => allowFormSubmission(2),
+                textFocus: _phoneTextFocus,
+                controller: _phoneTextEditingController,
+              ),
+              SignupForm(
+                stepNumber: "3",
+                fieldLabel: "Enter Email Id",
+                buttonLabel: "Send OTP",
+                hint: "admin@mail.com",
+                isActive: _stepNumber == 3 && (!sendOtpSuccess.value),
+                isOperationAllowedButNotActive: _stepNumber > 3,
+                buttonClickHandler: () {
+                  verifyUser();
+                },
+                buttonVisible: true,
+                textFocus: _emailTextFocus,
+                controller: _emailTextEditingController,
+              ),
+              ValueListenableBuilder<bool>(
+                valueListenable: sendOtpSuccess,
+                builder: (context, value, child) {
+                  return Visibility(
+                    visible: value,
+                    child: SignupForm(
+                      stepNumber: "4",
+                      fieldLabel: "Enter OTP",
+                      buttonLabel: "Verify OTP",
+                      hint: "1234",
+                      isActive: _stepNumber == 4,
+                      isOperationAllowedButNotActive: _stepNumber > 4,
+                      buttonVisible: true,
+                      buttonClickHandler: () {
+                        verifyOtp();
+                      },
+                      textFocus: _otpTextFocus,
+                      textInputAction: TextInputAction.done,
+                      controller: _otpTextEditingController,
+                    ),
+                  );
+                },
+              ),
+            ],
           ),
-        ],
+        ),
       ),
     );
   }
+
+  Future<void> verifyUser() async {
+    if(_formKey.currentState !=null && _formKey.currentState!.validate()){
+      HttpsCallable request = FirebaseFunctions.instance.httpsCallable('login');
+      try {
+        final result = await request.call(<String, dynamic>{
+          "kilvishId": _kilvishTextEditingController.text,
+          "email": _kilvishTextEditingController.text,
+          "phone": _phoneTextEditingController.text,
+        });
+        print("response is here ${result.data}");
+        if (result.data['success']) {
+          sendOtpSuccess.value = true;
+          _otpTextFocus.requestFocus();
+        } else {
+          Fluttertoast.showToast(
+              msg: result.data['message'],
+              toastLength: Toast.LENGTH_SHORT,
+              gravity: ToastGravity.CENTER,
+              backgroundColor: Colors.red,
+              textColor: Colors.white,
+              fontSize: 16.0
+          );
+        }
+      } on FirebaseFunctionsException catch (error) {
+        print("Exception $error");
```

Comment:
I think when we will send error from the backend (for the case when user has entered wrong email or phone), we need to show an error message to the user as well.